src/Sentry.EntityFramework/SentryQueryPerformanceListener.cs

```diff
@@ -0,0 +1,70 @@
+using System.Data.Common;
+using System.Data.Entity.Infrastructure.Interception;
+using Sentry.EntityFramework.Internals.Extensions;
+using Sentry.Extensibility;
+
+namespace Sentry.EntityFramework
+{
+    internal class SentryQueryPerformanceListener : IDbCommandInterceptor
+    {
+        internal const string SentryUserStateKey = "SentrySpanRef";
+        internal const string DbReaderKey = "db.query";
+        internal const string DbNonQueryKey = "db.execute";
+        internal const string DbScalarKey = "db.query-scalar";
+
+        private SentryOptions _options { get; }
+        private IHub _hub { get; }
+
+        internal SentryQueryPerformanceListener(IHub hub, SentryOptions options)
+        {
+            _hub = hub;
+            _options = options;
+        }
+
+        public void ReaderExecuting(DbCommand command, DbCommandInterceptionContext<DbDataReader> interceptionContext)
+            => CreateSpan(DbReaderKey, command.CommandText, interceptionContext);
+
+        public void ReaderExecuted(DbCommand command, DbCommandInterceptionContext<DbDataReader> interceptionContext)
+            => Finish(DbReaderKey, interceptionContext);
+
+        public void NonQueryExecuting(DbCommand command, DbCommandInterceptionContext<int> interceptionContext)
+            => CreateSpan(DbNonQueryKey, command.CommandText, interceptionContext);
+
+        public void NonQueryExecuted(DbCommand command, DbCommandInterceptionContext<int> interceptionContext)
+            => Finish(DbNonQueryKey, interceptionContext);
+
+        public void ScalarExecuting(DbCommand command, DbCommandInterceptionContext<object> interceptionContext)
+            => CreateSpan(DbScalarKey, command.CommandText, interceptionContext);
+
+        public void ScalarExecuted(DbCommand command, DbCommandInterceptionContext<object> interceptionContext)
+            => Finish(DbScalarKey, interceptionContext);
+
+        private void CreateSpan<T>(string key, string? command,
+            DbCommandInterceptionContext<T> interceptionContext)
+        {
+            if (_hub.GetSpan()?.StartChild(key, command) is { } span)
+            {
+                interceptionContext.AttachSpan(span);
+            }
+        }
+
+        /// <summary>
+        /// Finishes the span contained on interceptionContext.
+        /// </summary>
+        /// <typeparam name="T">(unused) The TResult from the Interception.</typeparam>
+        /// <param name="key">The key operation, used for logging.</param>
+        /// <param name="interceptionContext">The data that must contain a Span reference.</param>
+        private void Finish<T>(string key, DbCommandInterceptionContext<T> interceptionContext)
+        {
+            //Recover direct reference of the Span.
+            if (interceptionContext.GetSpanFromContext() is { } span)
+            {
+                span.Finish(interceptionContext.Exception is null ? SpanStatus.Ok : SpanStatus.InternalError);
```

Comment:
if `interceptionContext.Exception` isn't null then use `Finish(interceptionContext.Exception)`

```suggestion
                span.Finish(interceptionContext.Exception is null ? SpanStatus.Ok : SpanStatus.InternalError);
```